### PR TITLE
Fix vulnerabilities in VS extension

### LINKED ISSUE #5293: severity vulnerability reported in the package referenced in vs extension
### Describe the bug

> C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe /m /v:m /noAutoRsp /p:Configuration=Release /p:Version=0.62.0 /restore C:\github\typespec-azure\core\packages\typespec-vs\Microsoft.TypeSpec.VS.sln
MSBuild version 17.12.6+db5f6012c for .NET Framework
  Determining projects to restore...
C:\github\typespec-azure\core\packages\typespec-vs\src\Microsoft.TypeSpec.VS.csproj : error NU1902: Warning As Error: Package 'MessagePack' 2.5.124 has a known moderate severity vulnerability, https://github.com/advis
ories/GHSA-4qm4-8hg2-g2xm [C:\github\typespec-azure\core\packages\typespec-vs\Microsoft.TypeSpec.VS.sln]
C:\github\typespec-azure\core\packages\typespec-vs\src\Microsoft.TypeSpec.VS.csproj : error NU1903: Warning As Error: Package 'Microsoft.IO.Redist' 6.0.0 has a known high severity vulnerability, https://github.com/adv 
isories/GHSA-hq7w-xv5x-g34j [C:\github\typespec-azure\core\packages\typespec-vs\Microsoft.TypeSpec.VS.sln]
C:\github\typespec-azure\core\packages\typespec-vs\src\Microsoft.TypeSpec.VS.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 8.0.0 has a known high severity vulnerability, https://github.com/adviso 
ries/GHSA-8g4q-xg66-9fp4 [C:\github\typespec-azure\core\packages\typespec-vs\Microsoft.TypeSpec.VS.sln]
C:\github\typespec-azure\core\packages\typespec-vs\src\Microsoft.TypeSpec.VS.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 8.0.0 has a known high severity vulnerability, https://github.com/adviso 
ries/GHSA-hh2w-p6rv-4g7w [C:\github\typespec-azure\core\packages\typespec-vs\Microsoft.TypeSpec.VS.sln]
  Failed to restore C:\github\typespec-azure\core\packages\typespec-vs\src\Microsoft.TypeSpec.VS.csproj (in 1.43 sec).
 ELIFECYCLE  Command failed with exit code 1.

### Reproduction

open typespec vs extension

### Checklist

- [x] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [x] Check that there isn't already an issue that request the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/Microsoft/typespec/discussions).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by RodgeFu**
fix #5293

### CODE REVIEW

File: packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj [hunk 1]
```diff
@@ -7,6 +7,9 @@
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <LangVersion>Latest</LangVersion>
     <Nullable>Enable</Nullable>
+    <!-- Set this to 'all' explicitly to get a consistent behavior. It's the default value now in latest VS/.Net/NugetMgr, but maybe not in some pervious version -->
```
Comment (markcowl) [inline packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj:10]:
```suggestion
    <!-- Set this to 'all' explicitly to get a consistent behavior. It's the default value now in latest VS/.Net/NugetMgr, but maybe not in some previous version -->
```

File: packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj [hunk 2]
```diff
@@ -28,6 +31,10 @@
       Link="TextMate/typespec.tmLanguage" />
     <Content Update="@(Content)" IncludeInVSIX="true" CopyToOutputDirectory="PreserveNewest" />
     <Reference Include="System.ComponentModel.Composition" />
+    <!-- The 'MessagePack' is not used directly but referenced by nuget package 'Microsoft.VisualStudio.SDK', 'Microsoft.VisualStudio.Workspace' and 'Microsoft.VisualStudio.VSIntegration' -->
+    <!-- Have to add it explicitly here because the version used by 'Microsoft.VisualStudio...' report vulnerabilities now -->
+    <!-- TODO: remove this explicit reference after 'Microsoft.VisualStudio...' releases new version having 'MessagePack' updated to vulnerability free version-->
```
Comment (markcowl) [inline packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj:36]: Please file an issue for this and reference in the TODO

File: packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj [hunk 3]
```diff
@@ -40,12 +47,18 @@
     </Content>
   </ItemGroup>
   <ItemGroup>
+    <!-- The 'Microsoft.VisualStudio.Composition' is not used directly but referenced by nuget package 'Microsoft.VisualStudio.Workspace' -->
+    <!-- Have to add it explicitly here because otherwise 'Microsoft.VisualStudio.Workspace" will report warning NU1603 because its desired version not found and higher version will be used -->
+    <!-- TODO: remove this explicit reference after 'Microsoft.VisualStudio...' releases new version without the warning-->
```
Comment (markcowl) [inline packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj:52]: Please file an issue for this and reference in the TODO

File: packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj [hunk 4]
```diff
@@ -40,12 +47,18 @@
...
+    <PackageReference Include="Microsoft.VisualStudio.Composition" Version="17.12.18" />
     <!-- Use 17.0.x or latest 16.x if no 17.0.x for compatible API-->
-    <PackageReference Include="Microsoft.VisualStudio.SDK" Version="17.10.40171" ExcludeAssets="Runtime" />
-    <PackageReference Include="Microsoft.VisualStudio.Workspace" Version="16.3.43"
-      ExcludeAssets="Runtime" />
-    <PackageReference Include="Microsoft.VisualStudio.Workspace.VSIntegration" Version="16.3.43"
-      ExcludeAssets="Runtime" />
+    <PackageReference Include="Microsoft.VisualStudio.SDK" Version="17.12.40392" ExcludeAssets="Runtime" />
+    <!-- The 'Microsoft.VisualStudio.Shell.15.0' is not used directly but referenced by nuget package 'Microsoft.VisualStudio.Workspace.VSIntegration' -->
+    <!-- Have to add it explicitly here because otherwise 'Microsoft.VisualStudio.Workspace.VSIntegration" will report warning NU1603 because its desired version not found and higher version will be used -->
```
Comment (markcowl) [inline packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj:57]: Same.  It's OK if we use one issue to track

**Review by markcowl — CHANGES_REQUESTED** [review]
Some minor suggested changes, also:

- Do we need to release with these updates immediately? Should this target the release branch?
- What kind of testing do we or can we do to ensure the dependencies are valid?

File: packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj (hunk 2 above)
Comment (RodgeFu) [reply packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj:36]: #5368 filed to track the issue.

File: packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj (hunk 3 above)
Comment (RodgeFu) [reply packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj:52]: #5368 filed to track the issue.

File: packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj (hunk 4 above)
Comment (RodgeFu) [reply packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj:57]: #5368 filed to track the issue. thx.

**Comment by RodgeFu** [comment]
> Some minor suggested changes, also:
> 
> * Do we need to release with these updates immediately? Should this target the release branch?
> * What kind of testing do we or can we do to ensure the dependencies are valid?
      
* I think it should be fine to go with next normal release. But I am not expert in security and I know we are holding a high bar for security now. So please help to suggest or include any security expert to comment if needed. thanks.
* Almost all the functionalities in VS extensions now are provided by LSP implemented in compiler. I have tested the vs-extension to make sure the new version can talk with language server properly and major function works properly.

**Review by markcowl — APPROVED** [review]

**Review by markcowl — APPROVED** [review]

**Pull request merged**